### PR TITLE
generating interactive figures for attribution

### PR DESCRIPTION
**Pull request by joker2411**
## Description of the change

> updated the heatmap generated for markov model transition probabilities into an interactive figure as .html file. Ticket [Link](https://linear.app/rudderstack/issue/PRML-567/update-attribution-output-heatmap-image).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/profiles-pycorelib/attribution_model.py [hunk 1]
```diff
@@ -25,6 +28,7 @@ class AttributionModel(BaseModelType):
             "days_since_first_seen_var": { "type": "string" }, # model
             "first_seen_since": { "type": "integer" },
             "conversion_entity_var": { "type": "string" }, #model 
+            "enable_visualisation": { "type": "boolean" },
```
Comment (dpatchigolla) [inline src/profiles-pycorelib/attribution_model.py:31]: Can we set up the default from here itself, in the spec?

File: src/profiles-pycorelib/attribution_model.py [hunk 2]
```diff
@@ -88,23 +92,27 @@ def _generate_transition_counts(journey_list: List[List[Union[int, str]]],
...
-                        xticklabels=labels,
-                        yticklabels=labels)
-        ax.tick_params(labelsize=10)
-        ax.figure.set_size_inches((16, 10))
-        ax.set_ylabel("Previous Step")
-        ax.set_xlabel("Next Step")
-        ax.set_title(title)
-        plt.savefig(image_path)
+                          image_path, 
+                          title="Markov model transition probabilities"):
```
Comment (dpatchigolla) [inline src/profiles-pycorelib/attribution_model.py:96]: The title is no longer correct. Can you modify to "User journey map"

File: src/profiles-pycorelib/attribution_model.py [hunk 3]
```diff
@@ -152,13 +160,17 @@ def _get_removal_affects(transition_probs, labels, ignore_labels=["Start", "Drop
...
         transition_probabilities_converged = MultiTouchModels._converge(transition_probabilities, max_iters=500, verbose=False)
-        if attribution_reports_folder_path:
-            image_file = os.path.join(attribution_reports_folder_path, "markov_transition_probabilities.png")
-            MultiTouchModels._plot_transitions(transition_probabilities, labels, image_file)
+        if attribution_reports_folder_path and enable_visualisation:
+            if len(distinct_touches_list) <= MAXIMUM_TOUCHPOINTS_TO_VISUALIZE:
+                image_file = os.path.join(attribution_reports_folder_path, "markov_transition_probabilities.html")
+                MultiTouchModels._plot_transitions(transition_probabilities, labels, image_file)
+            else:
+                print("Too many touchpoints to visualise. Please use final attribution score table to visualize results.")
```
Comment (dpatchigolla) [inline src/profiles-pycorelib/attribution_model.py:173]: This should be self.logger.info, not in print.

Comment (dpatchigolla) [reply src/profiles-pycorelib/attribution_model.py:173]: Too many touchpoints to visualise. Please use final attribution score table to visualize results >

Skipping visualising the attribution model outputs as there are too many touchpoints. Visualisation is supported only when we have fewer than {MAXIMUM....} touchpoints

**Review by dpatchigolla — CHANGES_REQUESTED** [review]
minor suggestions on renaming

File: src/profiles-pycorelib/attribution_model.py (hunk 1 above)
Comment (joker2411) [reply src/profiles-pycorelib/attribution_model.py:31]: not available rn.

File: src/profiles-pycorelib/attribution_model.py [hunk 4]
```diff
@@ -282,6 +279,9 @@ def _convert_str_to_list(x):
         attribution_scores = self._get_first_touch_scores(filtered_df,
                                                          touch_point_var, 
                                                          conversion_var)
+        if attribution_scores.shape[0] > MAXIMUM_TOUCHPOINTS_TO_VISUALIZE:
```
Comment (dpatchigolla) [inline src/profiles-pycorelib/attribution_model.py:282]: Can you add `enable_visualisation` too in the check? So 
`if enable_visualisation and attribution_scores.shape[0] > MAXIMUM_TOUCHPOINTS_TO_VISUALIZE:`

This way, we don't print a log message saying we are skipping visualisation if the flag is already False.

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**